tests/test_reqhogger.py

```diff
@@ -0,0 +1,48 @@
+import os
+from unittest import TestCase
+from unittest.mock import patch
+
+from tinydb.storages import JSONStorage, MemoryStorage
+
+from hogger.reqhogger import RequestStorage
+
+
+class RequestStorageTestCase(TestCase):
+    def setUp(self):
+        self.sample_data = {"name": "Neil Armstrong", "year": 1969}
+        self.request_storage = RequestStorage()
+
+    def tearDown(self):
+        self.request_storage.clean()
+
+    @patch.dict("os.environ", {"HOGGER_PERSIST": "1"})
+    def test_instantiate_class_with_hogger_persist(self):
+        self.assertEqual(os.environ["HOGGER_PERSIST"], "1")
+        request_storage = RequestStorage()
+        self.assertTrue(isinstance(request_storage.storage.storage, JSONStorage))
```

Comment:
This is probably the culprit. You're creating a storage instance without explicitly closing it. We probably just want to use that in a context, like:

 ```python
    with RequestStorage() as request_storage:
        self.assertTrue(isinstance(request_storage.storage.storage, JSONStorage))
```

This _might_ do the trick, but I haven't tested it

Reply:
But alas, it doesn't.

The RequestStorage class doesn't reimplement the TinyDB storage (D'oh!). What we can do here is to enforce it manually after the test is complete.

Probably just doing
```python
request_storage.storage.close()
```
will be enough. If the assertion fails, we won't close the file, but that's an error anyway, so I think we're good.

Reply:
Good call @mawkee! Thanks 💯